### PR TITLE
Add Device Group env vars API support

### LINKED ISSUE #4658: Add Database column and API for updating device group settings env

### PR DESCRIPTION
**Pull request by Steve-Mcl**
closes #4658

### Route added
* `PUT` `/api/v1/applications/:applicationId/device-groups/:groupId/settings`
    * Expects body to contain an `env` array of objects

NOTE: Adding endpoint vs updating the `PUT /api/v1/applications/:applicationId/device-groups/:groupId` route
was a toss of a coin.  Instances overload the `PUT /api/v1/projects/:id` route allowing for the
settings to be updated in the same request.  However, devices have a separate route `PUT /api/v1/devices/:deviceId` route.
Since devices and device groups are closer in relation, I decided to follow suit. I did however structure this is a way that we can easily remove this endpoint and map the feature to the existing route if so desired

### Specified behaviour
* Only an owner role user can update the group settings
  * The RBAC `application:device-group:update` permission was re-used for this purpose
* The `env` array of objects is merged with the existing member devices only in the `live` API (when a device is requesting settings)
* Updating a device group model `settings` and saving will trigger all member devices to update their settings hash
* When the hash is computed, group env vars are merged with the device env vars before computing the hash (they are not persisted in the device model)

### Tests added
```
  Audit Log > Application
    ✔ Provides a logger for updating device group settings
```

```
  Application Device Groups API
    Update Device Settings
      ✔ Owner can update a device group settings (414ms)
      ✔ Merges new Group Env Var (351ms)
      ✔ Only updates a device settings hash if the group env var change affects the merged settings env (327ms)
      ✔ Member can not update a device group settings (403) (95ms)
      ✔ Non Member can not update a device group settings (404) (95ms)
```

## Related Issue(s)

Owner: #4658 
Parent: #4538

## Checklist

 - [x] I have read the [contribution guidelines](https://github.com/FlowFuse/flowfuse/blob/main/CONTRIBUTING.md)
 - [x] Suitable unit/system level tests have been added and they pass 
 - [ ] Documentation has been updated
    - [ ] Upgrade instructions
    - [ ] Configuration details
    - [ ] Concepts
 - [ ] Changes `flowforge.yml`?
    - [ ] Issue/PR raised on `FlowFuse/helm` to update ConfigMap Template
    - [ ] Issue/PR raised on `FlowFuse/CloudProject` to update values for Staging/Production

## Labels

 - [x] Includes a DB migration? -> add the `area:migration` label

### CODE REVIEW

File: forge/ee/db/controllers/DeviceGroup.js [hunk 1]
```diff
@@ -214,6 +243,58 @@ module.exports = {
         await app.db.models.Device.update({ DeviceGroupId: null }, { where: { id: deviceIds.removeList, DeviceGroupId: deviceGroup.id }, transaction })
     },
 
+    // updateSettings: async function (app, deviceGroup, settings, user) {
```
Comment (knolleary) [inline forge/ee/db/controllers/DeviceGroup.js:246]: Unused code? Something for the future?

**Review by knolleary — APPROVED** [review]
Looks good - one pending question about a commented out function. Approved pending resolution of the question.

File: forge/ee/db/controllers/DeviceGroup.js (hunk 1 above)
Comment (Steve-Mcl) [reply forge/ee/db/controllers/DeviceGroup.js:246]: Ah, that slipped through. Originally, I wrote this as a separate function but the tests revealed I need to restructure.

Will remove.

Comment (Steve-Mcl) [reply forge/ee/db/controllers/DeviceGroup.js]: resolved in [Remove unused code](https://github.com/FlowFuse/flowfuse/pull/4659/commits/f2aad783993211f47a918dd82d14a3951d8f181d)

**Pull request merged**